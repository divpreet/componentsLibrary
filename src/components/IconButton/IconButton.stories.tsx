import * as React from 'react';
import { action } from '@storybook/addon-actions';
import { Story } from '@storybook/react';
import { boolean, text, withKnobs } from '@storybook/addon-knobs';
import IconButton from './IconButton';

export default {
    title: 'Inputs / IconButton',
    component: IconButton,
    componentSubtitle: 'Displays a IconButton',
    decorators: [withKnobs]
};

const spacer = <span style={{ display: 'inline-block', marginRight: 16 }} />;

export const component: Story = () => {
    return (
        <div>
            <IconButton
                a11yId="1"
                icon={<span className="icon icon-cube-outline" />}
                text={text('text', 'Icon Button', '1')}
                ariaLabel={text('ariaLabel', 'Icon Button', '1')}
                disabled={boolean('disabled', false, '1')}
                onClickHandler={() => {
                    action('onClickHandler');
                }}
            />
            {spacer}
            <IconButton
                a11yId="123"
                icon={<span className="icon icon-earth" />}
                text={text('text', 'Icon Button', '123')}
                disabled={boolean('disabled', true, '123')}
                onClickHandler={() => {
                    action('onClickHandler');
                }}
            />
        </div>
    );
};

export const roundedDefault: Story = () => {
    return (
        <div>
            <IconButton
                a11yId="1"
                icon={<span className="icon icon-cube-outline" />}
                text={text('text', 'Icon Button', '1')}
                ariaLabel={text('ariaLabel', 'Icon Button', '1')}
                disabled={boolean('disabled', false, '1')}
                onClickHandler={() => {
                    action('onClickHandler');
                }}
                rounded
            />
        </div>
    );
};

export const roundedDisabled: Story = () => {
    return (
        <div>
            <IconButton
                a11yId="123"
                icon={<span className="icon icon-earth" />}
                text={text('text', 'Icon Button', '123')}
                disabled={boolean('disabled', true, '123')}
                onClickHandler={() => {
                    action('onClickHandler');
                }}
                rounded
            />
        </div>
    );
};

export const roundedPrimary: Story = () => {
    return (
        <div>
            <IconButton
                a11yId="123"
                icon={<span className="icon icon-earth" />}
                text={text('text', 'Icon Button', '123')}
                disabled={boolean('disabled', false, '123')}
                onClickHandler={() => {
                    action('onClickHandler');
                }}
                accent
                rounded
            />
        </div>
    );
};
