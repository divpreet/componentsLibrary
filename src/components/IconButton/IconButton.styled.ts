import { dclStyled } from '../Theme';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import * as sizes from '../../styles/dclSizes.json';

export const ButtonStyled = dclStyled(Button)`
    &.dclIconButton__root {
        height: ${sizes.spacing.factor4.s8};
        line-height: ${sizes.spacing.factor4.s8};
        cursor: pointer;
        padding: 0 ${sizes.spacing.factor4.s3};
        background: ${({ theme }) => theme.background.default};
        color: ${({ theme }) => theme.text.secondary};
        min-width: 0;
        width: ${sizes.spacing.factor4.s6};
        padding: 0 ${sizes.spacing.factor4.s4};

        &:hover {
            background-color: ${({ theme }) => theme.background.default};
            color: ${({ theme }) => theme.primary.main};
        }

        &:active {
            background: ${({ theme }) => theme.background.default};
            color: ${({ theme }) => theme.primary.main};
        }

        & .MuiButton-startIcon {
            margin: 0;
        }

        & .dclIconButton__text {
            text-indent: -999em;
            visibility: hidden;
            width: 0;
        }

        &.Mui-disabled {
            color: ${({ theme }) => theme.text.disabled};
            background-color: transparent;
            border-color: transparent;
        }
    }
`;

export const IconButtonStyled = dclStyled(IconButton)`
    &.dclIconButton__root {
        color: ${({ theme }) => theme.text.primary};
        background-color: transparent;
        width: ${sizes.spacing.factor4.s10};
        height: ${sizes.spacing.factor4.s10};

        &.Mui-disabled {
            color: ${({ theme }) => theme.text.disabled};
            background-color: transparent;
            border-color: transparent;
        }

        & .MuiIconButton-label > span.icon {
            font-size: ${sizes.spacing.factor4.s4};
        }

        &:hover {
            color: ${({ theme }) => theme.primary.main};
        }

        &.selected {
            color: ${({ theme }) => theme.primary.main};
        }

        &:active {
            color: ${({ theme }) => theme.primary.main};
        }

        & .dclIconButton__text {
            text-indent: -999em;
            visibility: hidden;
            width: 0;
        }
    }
    &.dclIconButton__primary {
        background-color: transparent;
        color: ${({ theme }) => theme.primary.main};
        &.Mui-disabled {
            color: ${({ theme }) => theme.text.disabled};
            background-color: transparent;
            border-color: transparent;
        }
        &:hover {
            color: ${({ theme }) => theme.primary.main};
        }
    }
`;
