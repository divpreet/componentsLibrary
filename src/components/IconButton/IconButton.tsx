import * as React from 'react';
import { MouseEventHandler } from 'react';
import { ButtonStyled, IconButtonStyled } from './IconButton.styled';

export interface IconButtonProps {
    a11yId?: string;
    ariaLabel?: string;
    className?: string;
    disabled?: boolean;
    icon?: React.ReactNode;
    onClickHandler: MouseEventHandler;
    tabIndex?: number;
    text: string;
    accent?: boolean;
    rounded?: boolean;
}

const IconButton: React.FC<IconButtonProps> = ({
    accent = false,
    a11yId,
    ariaLabel,
    className,
    disabled = false,
    icon,
    onClickHandler,
    tabIndex,
    text,
    rounded = false
}) => {
    if (rounded) {
        return (
            <IconButtonStyled
                aria-label={ariaLabel || text}
                className={className}
                classes={{
                    root: 'dclIconButton__root',
                    colorPrimary: 'dclIconButton__primary'
                }}
                color={accent ? 'primary' : 'inherit'}
                disabled={disabled}
                onClick={onClickHandler}
                id={a11yId}
                tabIndex={tabIndex}
                title={text}
            >
                {icon}
                <span className="dclIconButton__text">{text}</span>
            </IconButtonStyled>
        );
    }
    return (
        <ButtonStyled
            aria-label={ariaLabel || text}
            className={className}
            classes={{
                root: 'dclIconButton__root'
            }}
            color="inherit"
            disabled={disabled}
            disableElevation
            onClick={onClickHandler}
            id={a11yId}
            startIcon={icon}
            variant="text"
            tabIndex={tabIndex}
            title={text}
        >
            <span className="dclIconButton__text">{text}</span>
        </ButtonStyled>
    );
};

export default IconButton;
