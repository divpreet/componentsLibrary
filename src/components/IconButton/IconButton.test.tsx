import * as React from 'react';
import IconButton from './IconButton';
import { mount, shallow } from 'enzyme';

const onClickHandler = jest.fn();
const props = {
    a11yId: '23234',
    disabled: true,
    onClickHandler,
    text: 'text',
    icon: <span className="icon icon-earth" />
};

describe('IconButton', () => {
    let buttonProps = props;

    beforeEach(() => {
        buttonProps = {
            ...props
        };
    });

    it('renders correctly', () => {
        const component = shallow(<IconButton {...buttonProps} />).dive();
        expect(component).toMatchSnapshot();
    });

    test('should display with icon', () => {
        const component = mount(<IconButton {...buttonProps} />);
        const icon = component.find('icon-earth');

        expect(icon).toBeDefined();
    });

    test('should display rounded button with icon', () => {
        const component = mount(<IconButton {...buttonProps} rounded />);
        const icon = component.find('icon-earth');

        expect(icon).toBeDefined();
    });

    test('should display rounded accent button with icon', () => {
        const component = mount(<IconButton {...buttonProps} rounded accent />);
        const icon = component.find('icon-earth');

        expect(icon).toBeDefined();
    });

    it('should render hidden label', () => {
        const component = mount(<IconButton {...props} />);
        const text = component.find('.dclIconButton__text');

        expect(text.text()).toEqual(props.text);
    });

    it('should fire onClickHandler when clicked', () => {
        const component = shallow(<IconButton {...props} />);

        component.simulate('click');

        expect(onClickHandler.mock.calls.length).toEqual(1);
    });
});
