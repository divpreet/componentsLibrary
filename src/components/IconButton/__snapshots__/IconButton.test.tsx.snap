// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders correctly 1`] = `
.c0.dclIconButton__root {
  height: 32px;
  line-height: 32px;
  cursor: pointer;
  padding: 0 12px;
  background: #ffffff;
  color: #525965;
  min-width: 0;
  width: 24px;
  padding: 0 16px;
}

.c0.dclIconButton__root:hover {
  background-color: #ffffff;
  color: #0266C8;
}

.c0.dclIconButton__root:active {
  background: #ffffff;
  color: #0266C8;
}

.c0.dclIconButton__root .MuiButton-startIcon {
  margin: 0;
}

.c0.dclIconButton__root .dclIconButton__text {
  text-indent: -999em;
  visibility: hidden;
  width: 0;
}

.c0.dclIconButton__root.Mui-disabled {
  color: #707070;
  background-color: transparent;
  border-color: transparent;
}

<WithStyles(ForwardRef(Button))
  aria-label="text"
  className="c0"
  classes={
    Object {
      "root": "dclIconButton__root",
    }
  }
  color="inherit"
  disableElevation={true}
  disabled={true}
  id="23234"
  onClick={[MockFunction]}
  startIcon={
    <span
      className="icon icon-earth"
    />
  }
  theme={
    Object {
      "alerts": Object {
        "button": "#32373F",
        "error": "#BD2030",
        "errorBg": "#F7E4E6",
        "info": "#0266C8",
        "infoBg": "#E0F3FE",
        "success": "#2A7D2A",
        "successBg": "#DFF6DF",
        "warning": "#F3AD2E",
        "warningBg": "#FDEED3",
      },
      "background": Object {
        "default": "#ffffff",
        "paper": "#F5F5F5",
        "tertiary": "#EFEFEB",
      },
      "brandColours": Object {
        "default": "#007DC3",
        "secondary": "#20908B",
      },
      "divider": "#d1d1d1",
      "global": Object {
        "headerBg": "#32373F",
        "navigationBg": "#32373F",
        "navigationSelectedBg": "#525965",
        "navigationText": "#ffffff",
      },
      "highlight": Object {
        "blue": "#E0F3FE",
        "green": "#DFF6DF",
        "red": "#F7E4E6",
        "yellow": "#FDEED3",
      },
      "meaningful": Object {
        "blue": "#0266C8",
        "green": "#2A7D2A",
        "purple": "#9D4098",
        "red": "#BD2030",
        "yellow": "#F3AD2E",
      },
      "primary": Object {
        "main": "#0266C8",
      },
      "text": Object {
        "disabled": "#707070",
        "primary": "#32373F",
        "secondary": "#525965",
      },
      "typography": Object {
        "body1": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "body2": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
        "h1": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "h2": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "h3": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "h4": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "h5": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "h6": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "htmlFontSize": 14,
        "subtitle1": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "subtitle2": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
      },
    }
  }
  title="text"
  variant="text"
>
  <span
    className="dclIconButton__text"
  >
    text
  </span>
</WithStyles(ForwardRef(Button))>
`;
